src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -647,6 +647,9 @@ public Void caseTemplatizedType(TemplatizedType type) {
             emit("NodeList");
             return null;
           }
+          if (type.getDisplayName().equals("IThenable")) {
```

Comment:
Refactored. Btw, this is still quite yuki special case code. We talked with Alex about even bigger refactoring to have these guys be driven by a configuration file. The plan is to collect a bit more examples first.